### ISSUE TITLE
Change the theme of the refman and the stdlib

### ISSUE THREAD

**Issue by Zimmi48** [coq/coq#19976]
They should become adapted to the new Rocq Prover website.

**Comment by ppedrot** [comment]
Is there anybody working on this for the RC? If so it needs an assignee.

**Comment by mattam82** [comment]
Here you can see (temporarily) the proposed changes to the refman

https://rocq-prover.org/doc/master/refman/proofs/writing-proofs/proof-mode.html

**Comment by mattam82** [comment]
<img width="1149" alt="image" src="https://github.com/user-attachments/assets/f57e6562-2cb2-4679-857d-81ed2ff526fa" />

@BastienSozeau shouldn't the Rocq code examples be different as well as you modified the coqdoc style?

**Comment by Zimmi48** [comment]
The proposed theme mostly looks good to me, except for the orange background of the top-left corner. I would prefer using the blue and having the text in white as before. Or use a lighter orange there. I like the light orange of the tactics / commands.

**Comment by mattam82** [comment]
cc @BastienSozeau

**Comment by BastienSozeau** [comment]
@mattam82 @Zimmi48 I need you to harmonize the syntax theming across the docs and website.

We have different CSS classes generated and I have blind spots on what matches and what doesn't.

Could you do a mapping between this two lists of classes : 

List 1 (use for the refman and stdlib) : 

```
constructor
var
variable
definition
abbreviation
lemma
instance
projection
method
inductive
record
class
keyword

```

List 2 (use on the website [alectryon] and inline text) : 

```
Comment
Error
Generic
Keyword
Literal
Name
Operator
Other
Punctuation
Comment.Hashbang
Comment.Multiline
Comment.Preproc
Comment.PreprocFile
Comment.Single
Comment.Special
Generic.Deleted
Generic.Emph
Generic.Error
Generic.Heading
Generic.Inserted
Generic.Output
Generic.Prompt
Generic.Strong
Generic.Subheading
Generic.Traceback
Keyword.Constant
Keyword.Declaration
Keyword.Namespace
Keyword.Pseudo
Keyword.Reserved
Keyword.Type
Literal.Date
Literal.Number
Literal.String
Name.Attribute
Name.Builtin
Name.Class
Name.Constant
Name.Decorator
Name.Entity
Name.Exception
Name.Function
Name.Label
Name.Namespace
Name.Other
Name.Property
Name.Tag
Name.Variable
Operator.Word
Text.Whitespace
Literal.Number.Bin
Literal.Number.Float
Literal.Number.Hex
Literal.Number.Integer
Literal.Number.Oct
Literal.String.Affix
Literal.String.Backtick
Literal.String.Char
Literal.String.Delimiter
Literal.String.Doc
Literal.String.Double
Literal.String.Escape
Literal.String.Heredoc
Literal.String.Interpol
Literal.String.Other
Literal.String.Regex
Literal.String.Single
Literal.String.Symbol
Name.Builtin.Pseudo
Name.Function.Magic
Name.Variable.Class
Name.Variable.Global
Name.Variable.Instance
Name.Variable.Magic
Literal.Number.Integer.Long

```

**Comment by cpitclaudel** [comment]
The place to look at on the Alectryon side is this:

https://github.com/cpitclaudel/alectryon/blob/5a35676636766d2c6de8f1856a4bc168a3515ecb/alectryon/pygments_lexer.py#L320C1-L330C1

Can you point me to where the classes from the first list are computed / applied?

**Comment by BastienSozeau** [comment]
The first list is taken from the coqdoc.css 

Thoses classes are used for the reference manual and the stdlib 


https://github.com/coq/doc/blob/master/v9.0/refman/_static/coqdoc.css

**Comment by cpitclaudel** [comment]
Thanks. I see now that this issue says "closed" — is it actually resolved?

**Comment by Zimmi48** [comment]
The refman theme has been updated:

- https://rocq-prover.org/doc/v9.0/refman/index.html
- https://rocq-prover.org/doc/master/refman/index.html

But as far as I can see, it hasn't been the case of the corelib / stdlib doc:

- https://rocq-prover.org/doc/v9.0/corelib/index.html
- https://rocq-prover.org/doc/master/corelib/index.html
- https://rocq-prover.org/doc/v9.0/stdlib/index.html
- https://rocq-prover.org/doc/master/stdlib/index.html

Nor of the refman-stdlib BTW:

- https://rocq-prover.org/doc/v9.0/refman-stdlib/index.html
- https://rocq-prover.org/doc/master/refman-stdlib/index.html

I see that there was a PR on the stdlib repo linked to this issue that hasn't been merged yet:
https://github.com/coq/stdlib/pull/87

In there, it was also suggested to first open a similar PR for corelib and test it there. Was this done?

Should we reopen the issue until all of this is completed? @mattam82 @BastienSozeau

**Comment by mattam82** [comment]
Not, it’s not resolved yet

**Comment by mattam82** [comment]
As far as I can tell, Alectryon know very little about the identifiers it sees, the mapping would be:

```
constructor, var, variable, definition, abbreviation, lemma, instance, projection,
method, inductive, record, class -> no highlight or "nf" class in html for the 
   constant definition (as in Definition <span class="nf">foo<span> 
tactic -> tacn: Name.builtin
keyword -> gallina-constants: Keyword.Type
```

**Comment by Zimmi48** [comment]
Still not fully resolved (until similar PR for corelib).

**Comment by Zimmi48** [comment]
Reopening one last time because the stdlib refman (https://rocq-prover.org/doc/master/refman-stdlib) still needs to be updated as well.

**Comment by cpitclaudel** [comment]
> The variables/constants/inductives/constructor distinction is not available to Alectryon (yet).

That's right.  My hope is that we can get this information from LSP soon instead of having a custom Pygments parser.
@mattam82 your mapping looks good for what we currently have.

**Comment by Zimmi48** [comment]
@proux01 Any idea why https://rocq-prover.org/doc/v9.0/stdlib/index.html is still using the old theme even if https://rocq-prover.org/doc/master/stdlib/index.html is on the new one?

**Comment by proux01** [comment]
Likely because stdlib is pinned in the v9.0 branch in file ci-basic-overlay.sh. I guess that pin should be updated @ppedrot ?

**Comment by Zimmi48** [comment]
Right, I forgot about this despite having done the previous bump myself! (#20105)

**Comment by mattam82** [comment]
> Reopening one last time because the stdlib refman (https://rocq-prover.org/doc/master/refman-stdlib) still needs to be updated as well.

Right!

**Comment by mattam82** [comment]
Done here: https://github.com/coq/stdlib/pull/112

**Comment by mattam82** [comment]
Fixed by coq/stdlib#112

**Comment by proux01** [comment]
@mattam82 you probably want to update the stdlib pin in the v9.0 branch before marking this as completed
